src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```
@@ -24,6 +24,7 @@ The .NET Foundation licenses this file to you under the MIT license.
     <LinkerFlavor Condition="'$(LinkerFlavor)' == '' and '$(_linuxLibcFlavor)' == 'bionic'">lld</LinkerFlavor>
     <LinkerFlavor Condition="'$(LinkerFlavor)' == '' and '$(_targetOS)' == 'linux'">bfd</LinkerFlavor>
     <IlcDefaultStackSize Condition="'$(IlcDefaultStackSize)' == '' and '$(_linuxLibcFlavor)' == 'musl'">1572864</IlcDefaultStackSize>
+    <UseSystemZlib Condition="'$(_targetOS)' == 'ios' or '$(_targetOS)' == 'iossimulator' or '$(_targetOS)' == 'tvos' or '$(_targetOS)' == 'tvossimulator' or '$(_targetOS)' == 'maccatalyst' or '$(_targetOS)' == 'android'">true</UseSystemZlib>
```

```suggestion
    <UseSystemZlib Condition="!Exists($(IlcSdkPath)libz.a)">true</UseSystemZlib>
```
This may be even better. It should just work for the distro source build packages that should not be bundling libz.a library.

---

Do you plan to take this or the other suggestion to simplify the condition?

---

Yep! I just want to see if I get more feedback before I cause the CI to restart. Meanwhile I'm running the other azp runs.

---

I'm opting for the previous suggestion, as it is explicitly indicating which platforms to condition for. If I use Exists here, I am worried that more platforms will end up unexpectedly getting captured by the condition, and that does not match the other similar conditions elsewhere.

---

> other similar conditions elsewhere.

FWIW, we have a few Exists conditions in these target files and I think you will end up with the Exists condition to handle the non-portable build case. This is fine for now.

---

The other suggestion didn't work unfortunately:
>     <UseSystemZlib Condition="'$(_IsiOSLikePlatform)' == 'true' or '$(_targetOS)' == 'android'">true</UseSystemZlib>

Android, ios and tvos broke because _IsiOSLikePlatform was not available: https://dev.azure.com/dnceng-public/public/_build/results?buildId=723024&view=logs&j=11fc76eb-f2d5-5967-e9b4-f54329587f75&t=5d22afdd-fd55-5fd8-7ab1-37ebf23474bd

I'll use Exists.

---

How can `_IsiOSLikePlatform` be not available here? It is used in number of places in this file.

I think you have misdiagnosed the build error.

---

Yes, I misdiagnosed it. I am getting the same error with Exists.

---

I think I found the root cause, i added some exdtra `zlib` libraries in the cmake files but they needed to be conditioned.